src/main/java/frc/robot/commands/Test.java

```diff
@@ -20,38 +21,51 @@ public Test() {
     // eg. requires(chassis);
   }
 }*/
-public class Test extends CommandGroup {
+public class Test extends Command {
   M_I2C i2c = new M_I2C();// setup the i2c interface
   PixyPacket pkt = i2c.getPixy();// create a pixy packet to hold data
 
-  public void centerOnObject() {
+  public Test() {
+    super();
+    requires(Robot.m_drivetrain);
+
+  }
+  // public void centerOnObject() {
+
+  @Override
+  protected void execute() {
+
     if (pkt.x != -1) {// if data is exist
       if (pkt.x < .48 || pkt.x > .52) {
         // and the 'object', whatever it may be is not in the center
         // the code on the arduino decides what object to send
         while (pkt.x < .48 || pkt.x > .52) {// while it is not center
-
+          if (pkt.x > .52) {// if its on the right side of robot, turn right
+            Robot.m_drivetrain.drive(.2, -.2);
+            SmartDashboard.putString("direction", "right");
+          }
           if (pkt.x < .48) {// if its on the left side of robot, turn left
-            /*
-             * drive.setLDrive(-0.2);//this is our left side of tank drive
-             * drive.setRDrive(0.2);//you drive code might differ
-             */
-
+            SmartDashboard.putString("direction", "left");
             Robot.m_drivetrain.drive(-.2, .2);
           }
-          if (pkt.x > .52) {// if its on the right side of robot, turn right
-            /*
-             * drive.setLDrive(0.2); drive.setRDrive(-0.2);
-             */
-            Robot.m_drivetrain.drive(.2, -.2);
+
+          if (pkt.x > .48 && pkt.x < .52) {// while centered, stop
```

Comment:
You got the condition right but you want this farther down the file. 

The issue is that the data inside `pkt` only gets updated on line 59 when you call `pkt = i2c.getPixy()` so this condition will never be true.

If you put it right after line 59 it would work. You can do even one better though by putting it just outside the `while` loop. You don't even need the if statement then which is convenient. The `while` loop will keep running until the condition `pkt.x < .48 || pxt.x > .52` is true. So when it stops looping that's exactly when you want to turn off the motors. So something like
```
while(pkt.x < .48 || pkt.x > .52)
{
// handle centering the robot
}
Robot.m_driveTrain.drive(0,0);
```

Will cover when we finish centering. Conveniently it will _also_ cover the case that `pkt.y == -1` that you've handled because the `break` statement there means to stop the `while` loop even though the condition may still be true.

Reply:
If the second half of that doesn't make sense don't worry about it. We can chat about it a bit more on the next phase of pixy stuff. Just move this `if` and the code inside it down below the `pky = i2c.getPixy()` and it should start working.